include/hirediscommand.h

```diff
@@ -195,15 +196,15 @@ namespace RedisCluster
         
         redisReply* process()
         {
-            redisReply *reply;
+            redisReply *reply = nullptr;
             typename Cluster::SlotConnection con = cluster_p_->getConnection( key_ );
             typename Cluster::HostConnection hcon = { "", NULL };
             string host, port;
-            
+
             reply = processHiredisCommand( con.second );
+            HiredisProcess::checkCritical(reply, false, "", con.second);
             cluster_p_->releaseConnection( con );
```

Comment:
Hi just changed the order so I can pass the connection as parameter in checkcritical().
But you're right, I forgot that checkCritical can throw and then we don't release the connection.

I would then suggest something a bit like : 
 http://pastebin.com/JcVskCVX